pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -178,10 +180,13 @@ protected BasePartitionUpsertMetadataManager(String tableNameWithType, int parti
     }
     _serverMetrics = ServerMetrics.get();
     _logger = LoggerFactory.getLogger(tableNameWithType + "-" + partitionId + "-" + getClass().getSimpleName());
-    if (_metadataTTL > 0) {
+    if (isTTLEnabled()) {
+      Preconditions.checkState(_enableSnapshot, "Upsert TTL must have snapshot enabled");
```

Comment:
This adds a requirement for `deletedKeysTTL` to have snapshot-enabled which was not originally present. Can we keep check for deletedKeysTTL separate from `isTTLEnabled`?

Reply:
I actually tried to make two TTL types more unified, as I see many places check either one or two, and seems pretty error prone.

as to here, I think the two checks are true for both metadata and deleted TTL, right?

Reply:
For deletedKeysTTL we don't necessarily need snapshot to be enabled.

Reply:
thanks for the explanation offline, and I adjusted the checks here so that snapshot is required by metadataTTL only.